database/node.go

```diff
@@ -160,13 +158,38 @@ func (m *NodeManager) SetNodeInfo(server dqlite.NodeInfo) error {
 // WithLogFuncOption returns a Dqlite application Option that will proxy Dqlite
 // log output via this factory's logger where the level is recognised.
 func (m *NodeManager) WithLogFuncOption() app.Option {
+	var (
+		tracingEnabled   = m.cfg.QueryTracingEnabled()
+		tracingThreshold = m.cfg.QueryTracingThreshold()
+	)
+
 	return app.WithLogFunc(func(level client.LogLevel, msg string, args ...interface{}) {
```

Comment:
Probably a trivial difference, but I would return a different func altogether based on whether tracing is enabled. Then you don't have to check for each log message.